src/codec/decoders.rs

```diff
@@ -0,0 +1,23 @@
+use vector_core::event::Event;
+
+pub use crate::config::decoding::{Decoding, DecodingConfig};
+
+pub enum Decoder {
+    Utf8,
```

Comment:
I'd advocate for just supporting `json` and `syslog` for now. I think actually decoding as a character set will be a much bigger change as vector currently assumes utf8 everywhere it interprets bytes as a string. I think supporting alternate character sets will be useful in the future, but that it is outside the scope of what we should try to support here.

Reply:
I agree with you. Since no actual decoder is implemented yet, I just threw in some variants as a placeholder, since an empty enum can't be instantiated 😄